### PR TITLE
Bulktransfer improve

### PR DESCRIPTION
**Pull request by walaoaaa1234**
#5992 
#5987 #5988 #5994
# TODO
 - [x] DOC

### CODE REVIEW

**Comment by Jcolomar** [comment]
Lets test

File: pokemongo_bot/cell_workers/pokemon_optimizer.py [hunk 1]
```diff
@@ -699,13 +699,15 @@ def transfer_pokemon(self, transfer):
             4: 'ERROR_POKEMON_IS_EGG',
             5: 'ERROR_POKEMON_IS_BUDDY'
         }
-        pokemons = list(transfer)
         if self.config_bulktransfer_enabled and len(pokemons) > 1:
```
Comment (MerlionRock) [inline pokemongo_bot/cell_workers/pokemon_optimizer.py]: var pokemons not found. you should change to len(transfer)

File: pokemongo_bot/cell_workers/pokemon_optimizer.py [hunk 2]
```diff
@@ -699,13 +699,15 @@ def transfer_pokemon(self, transfer):
...
             while len(pokemons) > 0:
```
Comment (MerlionRock) [inline pokemongo_bot/cell_workers/pokemon_optimizer.py]: Same problem. All pokemons should change to transfer

File: pokemongo_bot/cell_workers/pokemon_optimizer.py [hunk 3]
```diff
@@ -699,13 +699,15 @@ def transfer_pokemon(self, transfer):
...
+                action_delay(self.config_action_wait_min, self.config_action_wait_max)
                 pokemon_ids = []
                 count = 0
+                transfered = []
                 while len(pokemons) > 0 and count < self.config_max_bulktransfer:
```
Comment (MerlionRock) [inline pokemongo_bot/cell_workers/pokemon_optimizer.py]: len(pokemons) -> len(transfer)

**Comment by MerlionRock** [comment]
@walaoaaa1234 see my comments~

**Comment by walaoaaa1234** [comment]
#5998

**Pull request merged**